### ISSUE TITLE
=ToFormattedString ?

### ISSUE THREAD

**Issue by blunck2** [bazaarvoice/jolt#440]
Hi Milo-

I'm using Jolt to transform JSON and it works very well in situations where you need to restructure the document.  I've run into a situation and have an approach to how we could potentially solve it and I was hoping you could share your insight on the approach and if it's achievable given the present design of Jolt.

I receive a date that is formatted "YYYY-MM-DD" and I need to translate it into "MM/DD/YYYY".  We presently achieve this by extending Transform.  The downside to this approach is tight coupling between our code and the underlying JSON documents and Jolt transformation.  Our DateTransform has to have knowledge of which JSON fields to transform and I'd prefer to leave all of that knowledge in our Jolt configuration.

I was thinking that using regular expressions and capture groups could provide a really powerful capability for Jolt.

Wouldn't it be great if in modify-overwrite-beta you could to a =ToFormattedString where you express a regular expression with capture groups and then also express the resulting text?  Something like the following:

```
"operation": "modify-overwrite-beta",
    "spec": {
        "person" : {
             "birthday": "=toFormattedString('^(19|20)(\d\d)[- /.](0[1-9]|1[012])[- /.](0[1-9]|[12][0-9]|3[01])$', '$3/$4/$1$2')"
        }
    }
}
```

I looked through com.bazaarvoice.jolt.modifier.function.Objects where the other modify-overwrite-beta functions are implemented (e.g. =toString, =toLong, ...).  I don't see any easy way to introduce parameters/arguments to the functions.  How difficult do you think it would be permit parameterized functions?

-c

**Comment by milosimpson** [comment]
The Modifier Function stuff is pretty terrible.   At the time I couldn't think of a good way to do it, but we needed the functionality.

I now have a better way of doing it (annotations and reflection) but I won't be able to work on it till November. 

A) I would like to make it easier for people / Java Developers to write their own functions.   I.e. you would be able to specify a Class name / method in the Modify Spec to use as a "function" name.
B) I personally hate Regex, or rather I personally hate debugging big stupid regexes, so I am really hesitant to offer "here have some regex" OOTB.   I will just get endless questions.
C) That is why I lean towards the OOTB functions being very "simple".

Basically what I am saying is, if you know enough to what to use the Java Regex implementation to solve a problem, you are the the type of person / user that can write you own Function to do.

To everyone else it would just be a loaded gun.
